flink-core/src/main/java/org/apache/flink/api/common/eventtime/CombinedWatermarkStatus.java

```diff
@@ -45,7 +45,7 @@ public long getCombinedWatermark() {
     }
 
     public boolean isIdle() {
-        return idle;
+        return (!partialWatermarks.isEmpty() || combinedWatermark > Long.MIN_VALUE) && idle;
```

Comment:
Couldn't we move the additional checks to `updateCombinedWatermark` to your newly added block? See suggestion below.